web/src/pages/Courts/CourtDetails/StakePanel/StakeWithdrawButton.tsx

```diff
@@ -97,86 +101,214 @@ const StakeWithdrawButton: React.FC<IActionButton> = ({
+              if (status) {
+                setPopupStepsState({
+                  items: getStakeSteps(
+                    isWithdraw ? StakeSteps.WithdrawConfirmed : StakeSteps.StakeConfirmed,
+                    amount,
+                    theme,
+                    approvalHash,
+                    hash
+                  ),
+                  current: 1,
+                });
+                setIsSuccess(true);
+              } else
+                setPopupStepsState({
+                  items: getStakeSteps(
+                    isWithdraw ? StakeSteps.WithdrawFailed : StakeSteps.StakeFailed,
+                    amount,
+                    theme,
+                    approvalHash,
+                    hash
+                  ),
+                  current: 1,
+                });
+            });
+          })
+          .catch((err) => {
+            setPopupStepsState({
+              items: getStakeSteps(
+                isWithdraw ? StakeSteps.WithdrawFailed : StakeSteps.StakeFailed,
+                amount,
+                theme,
+                approvalHash,
+                undefined,
+                err
+              ),
+              current: 1,
+            });
+          });
+      }
     },
-  };
+    [setStake, setStakeConfig, publicClient, amount, theme, action]
+  );
+
+  const handleClick = useCallback(() => {
+    setIsPopupOpen(true);
+    if (ActionType.allowance && isAllowance && increaseAllowanceConfig && publicClient) {
+      setPopupStepsState({
+        items: getStakeSteps(StakeSteps.ApproveInitiate, amount, theme),
+        current: 0,
+      });
+
+      increaseAllowance(increaseAllowanceConfig.request)
+        .then(async (hash) => {
+          setPopupStepsState({
+            items: getStakeSteps(StakeSteps.ApprovePending, amount, theme, hash),
+            current: 0,
+          });
+
+          await publicClient
+            .waitForTransactionReceipt({ hash, confirmations: 2 })
+            .then(async (res: TransactionReceipt) => {
+              const status = res.status === "success";
+              if (status) {
+                await refetchAllowance();
+                const refetchData = await refetchSetStake();
+
+                handleStake(refetchData.data, hash);
+              } else
+                setPopupStepsState({
+                  items: getStakeSteps(StakeSteps.ApproveFailed, amount, theme, hash),
+                  current: 0,
+                });
+            });
+        })
+        .catch((err) => {
+          setPopupStepsState({
+            items: getStakeSteps(StakeSteps.ApproveFailed, amount, theme, undefined, undefined, err),
+            current: 0,
+          });
+        });
+    } else {
+      handleStake();
+    }
+  }, [
+    increaseAllowance,
+    increaseAllowanceConfig,
+    handleStake,
+    isAllowance,
+    theme,
+    publicClient,
+    amount,
+    refetchAllowance,
+    refetchSetStake,
+  ]);
```

Comment:
Thought about it, but this logic is tightly linked to the component update. If it were to be abstracted, it would result in the same file. so there is no real benefit here.